src/Symfony/Component/Messenger/EnvelopeReaderInterface.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger;
+
+/**
+ * A Messenger protagonist aware of the message envelope and its content.
+ *
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ *
+ * @experimental in 4.1
+ */
+interface EnvelopeReaderInterface
```

Comment:
Wouldn't `EnvelopeAwareInterface` makes more sense? I don't have strong opinions on this one though to be honest :)

Reply:
No preference either.

Reply:
I'd prefer `EnvelopeAwareInterface` too.